pkg/ccl/backupccl/show.go

```diff
@@ -107,9 +134,53 @@ func showBackupPlanHook(
 				}
 			}
 		}
-		return nil
-	}
-	return fn, header, nil, nil
+	},
+}
+
+var backupShowerRanges = backupShower{
+	header: sqlbase.ResultColumns{
+		{Name: "start_pretty", Typ: types.String},
+		{Name: "end_pretty", Typ: types.String},
+		{Name: "start_key", Typ: types.Bytes},
+		{Name: "end_key", Typ: types.Bytes},
+	},
+
+	fn: func(desc BackupDescriptor, resultsCh chan<- tree.Datums) {
+		for _, span := range desc.Spans {
+			resultsCh <- tree.Datums{
```

Comment:
Since we are redoing this, might as well do it correctly for all of them: plumb a ctx down here and do the send in a select that is also doing a recv on ctx.Done. Or maybe we definitely don't need that? Seems like it wouldn't hurt and isn't hard.

Reply:
Given that these things aren't emitted incrementally, I'm going to adjust the shower interface to return a slice of tree.Datums and then do it properly in the hook fn. Fix incoming.

Reply:
Ok, fixed. PTAL.